src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -433,7 +433,7 @@ private function readProperty($zval, $property)
      */
     private function getReadAccessInfo($class, $property)
     {
-        $key = $class.'..'.$property;
+        $key = $this->sanitizeAnonymousClassName($class).'..'.$property;
```

Comment:
since this is perf critical, I suggest inline here, and use the fastest, which is rawurlencode. This might look like:
`rawurlencode($class)`
or
`false !== strpos($class, '@') ? rawurlencode($class) : $class`